src/main/java/fi/metatavu/famifarm/rest/interceptors/FacilityInterceptor.java

```diff
@@ -26,19 +27,28 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
         UriInfo uriInfo = requestContext.getUriInfo();
         String path = uriInfo.getPath();
         MultivaluedMap<String, String> pathParameters  = uriInfo.getPathParameters();
-        Optional<String> facility = pathParameters.get(FACILITY).stream().findFirst();
+        List<String> facilityParams = pathParameters.get(FACILITY);
+        CollectionUtils.filter(facilityParams, PredicateUtils.notNullPredicate());
+
+        if (facilityParams == null && Arrays.asList(WHITELISTED_PATHS).contains(path)) {
+            return;
+        }
+
+        if (facilityParams == null) {
```

Comment:
Doesn't this check include all the conditions of previous one + some extra?

Reply:
Resolved